app/server/lib/AuditLogger.ts

```diff
@@ -325,6 +354,16 @@ export class AuditLogger implements IAuditLogger {
       await redis.quitAsync();
     }
   }
+
+  private _track(prom: Promise<any>) {
+    this._createdPromises.push(prom);
+    return prom.finally(() => {
+      const index = this._createdPromises.indexOf(prom);
+      if (index !== -1) {
+        this._createdPromises.splice(index, 1);
+      }
```

Comment:
Agree, it is possible for auditing to get quite busy, and in combination with a slow log consumer the number of promises could get quite high and `indexOf` a little inefficient.

We do have a pending task to move audit message delivery to a queue, to insulate ourselves a bit again slow consumers, so all this will change again for installations with Redis available, but it is still worth it for regular instances.